test/integration/trackedProgresses.js

```diff
@@ -0,0 +1,457 @@
+      expect(data).to.have.property("id").that.is.a("string");
+      expect(data).to.have.property("type").that.equals("user");
+      expect(data).to.have.property("userId").that.equals(userId1);
+      expect(data).to.have.property("currentlyTracked").that.equals(true);
+      expect(data).to.have.property("frequency").that.equals(1);
+      expect(data).to.have.property("createdAt").that.is.a("string").and.satisfy(isISOString);
+      expect(data).to.have.property("updatedAt").that.is.a("string").and.satisfy(isISOString);
+    });
+
+    it("throws 409 if tracked progress document already exists for the same user", async function () {
+      const response = await chai
+        .request(app)
+        .post("/tracked-progresses")
+        .set("cookie", `${cookieName}=${superUserToken}`)
+        .send({
+          ...trackedProgressUserDataForPost,
+          userId: userId0,
+        });
+      expect(response).to.have.status(409);
+      expect(response.body).to.be.an("object");
+      expect(response.body).to.have.property("message").that.equals("Resource is already being tracked.");
+    });
+
+    it("stores the tracked progress document for task", async function () {
+      const response = await chai
+        .request(app)
+        .post("/tracked-progresses")
+        .set("cookie", `${cookieName}=${superUserToken}`)
+        .send({
+          ...trackedProgressTaskDataForPost,
+          taskId: taskId1,
+        });
+      expect(response).to.have.status(201);
+      expect(response.body).to.be.an("object");
+      expect(response.body).to.have.property("message").that.equals("Resource created successfully.");
+      expect(response.body).to.have.property("data").that.is.an("object");
+
+      const data = response.body.data;
+      expect(data).to.have.keys(["id", "type", "taskId", "currentlyTracked", "frequency", "createdAt", "updatedAt"]);
+      expect(data).to.have.property("id").that.is.a("string");
+      expect(data).to.have.property("type").that.equals("task");
+      expect(data).to.have.property("taskId").that.equals(taskId1);
+      expect(data).to.have.property("currentlyTracked").that.equals(true);
+      expect(data).to.have.property("frequency").that.equals(2);
+      expect(data).to.have.property("createdAt").that.is.a("string").and.satisfy(isISOString);
+      expect(data).to.have.property("updatedAt").that.is.a("string").and.satisfy(isISOString);
+    });
+
+    it("throws 409 if tracked progress document already exists for the same task", async function () {
+      const response = await chai
+        .request(app)
+        .post("/tracked-progresses")
+        .set("cookie", `${cookieName}=${superUserToken}`)
+        .send({
+          ...trackedProgressTaskDataForPost,
+          taskId: taskId0,
+        });
+      expect(response).to.have.status(409);
+      expect(response.body).to.be.an("object");
+      expect(response.body).to.have.property("message").that.equals("Resource is already being tracked.");
```

Comment:
I think this can be done like this
```js
expect(response.body.message).to.be.equal("Resource is already being tracked.");
```
it would automatically fail if the message property is not present so checking for the property sounds redundsnt